src/GoTrueClient.ts

```diff
@@ -2257,67 +2265,71 @@ export default class GoTrueClient {
    * {@see GoTrueMFAApi#listFactors}
    */
   private async _listFactors(): Promise<AuthMFAListFactorsResponse> {
-    const {
-      data: { user },
-      error: userError,
-    } = await this._getUser()
-    if (userError) {
-      return { data: null, error: userError }
-    }
+    return this._acquireLock(-1, async () => {
```

Comment:
why is it necessary to acquire the lock for listing factors?

Reply:
Good question! It's because of `_getUser` -- I did this somewhat mechanically. But yeah I can re-do it with calling `getUser` and it wouldn't be necessary! Thanks for spotting it!